src/test/java/org/assertj/core/internal/files/Files_assertSameContentAs_Test.java

```diff
@@ -115,4 +122,29 @@ public void should_fail_if_files_do_not_have_equal_content() throws IOException
     }
     failBecauseExpectedAssertionErrorWasNotThrown();
   }
+
+  @Test
+  public void should_pass_if_files_are_binary_identical() throws IOException {
+    unMockedFiles.assertSameContentAs(someInfo(), createFileWithNonUTF8Character(), createFileWithNonUTF8Character());
+  }
+
+  @Test
+  public void should_fail_if_files_are_not_binary_identical() throws IOException {
+    Throwable throwable = catchThrowable(() -> unMockedFiles.assertSameContentAs(someInfo(), createFileWithNonUTF8Character(), expected));
```

Comment:
Ok, I change that.

I make the PR for the 2.x branch or you will cherry-pick the commits ?